framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4095,7 +4114,18 @@ VkResult VulkanReplayConsumerBase::OverrideQueueSubmit(PFN_vkQueueSubmit
     // tracked.
     if ((!have_imported_semaphores_) && (options_.surface_index == -1) && (!options_.dumping_resources))
     {
-        result = func(queue_info->handle, submitCount, submit_infos, fence);
+        result = swapchain_->QueueSubmit(func,
+                                         queue_info,
+                                         submitCount,
+                                         submit_infos,
+                                         submit_info_data,
+                                         fence_info,
+                                         instance_info,
+                                         instance_table,
+                                         device_info,
+                                         device_table,
+                                         application_.get(),
```

Comment:
Can't we just put this in the swapchain options since the swapchain is device specific and we should already have all this info during creation?